src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -35,11 +37,16 @@ std::string ColumnBetweenTableScanImpl::description() const { return "ColumnBetw
 void ColumnBetweenTableScanImpl::_scan_non_reference_segment(
     const BaseSegment& segment, const ChunkID chunk_id, PosList& matches,
     const std::shared_ptr<const PosList>& position_filter) const {
-  // Select optimized or generic scanning implementation based on segment type
-  if (const auto* dictionary_segment = dynamic_cast<const BaseDictionarySegment*>(&segment)) {
-    _scan_dictionary_segment(*dictionary_segment, chunk_id, matches, position_filter);
+  const auto ordered_by = _in_table->get_chunk(chunk_id)->ordered_by();
+  if (ordered_by && ordered_by->first == _column_id) {
+    _scan_sorted_segment(segment, chunk_id, matches, position_filter, ordered_by->second);
   } else {
-    _scan_generic_segment(segment, chunk_id, matches, position_filter);
+    // Select optimized or generic scanning implementation based on segment type
+    if (const auto* dictionary_segment = dynamic_cast<const BaseDictionarySegment*>(&segment)) {
+      _scan_dictionary_segment(*dictionary_segment, chunk_id, matches, position_filter);
+    } else {
```

Comment:
My guess is that the performance regression for Q6 comes from not exploiting the dictionary any more. Instead, the binary search seems to decompress all values it looks at, which will be slower the fewer distinct values we have.

Reply:
That might be true for now but that should be no issue once we fix the iterators. Do you see this as an actual issue for this PR?

Reply:
Mmh, I originally thought that we should do the binary search on the dictionary, not on the values. However, I just realized that this does not necessarily reduce the number of string comparisons for input segments that are already filtered.

Could you have a look if this is affected by the problem and if the iterator trait hack helps here?